### PR TITLE
Flux pulses coupler experiments

### PR DESCRIPTION
**Pull request by Jacfomg**
Propagates #691 coupler spec routines plus some fixes/improvements

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: tests/test_protocols.py [hunk 1]
```diff
@@ -23,29 +23,34 @@
...
 
 
 def generate_runcard_single_protocol():
-    actions = yaml.safe_load(PATH_TO_RUNCARD.read_text(encoding="utf-8"))
-    with open(PATH_TO_RUNCARD) as file:
-        actions = yaml.safe_load(file)
-    for action in actions["actions"]:
-        card = {"actions": [action], "targets": list(PLATFORM.qubits)}
-        yield card
+    for runcard in [PATH_TO_RUNCARD_COUPLERS, PATH_TO_RUNCARD]:
```
Comment (GabrielePalazzo) [inline tests/test_protocols.py:34]: Should we move the for loop to a pytest parametrize fixture? If only one of the two runcards fails the test we know which one it is.

Comment (Jacfomg) [reply tests/test_protocols.py:34]: @andrea-pasquale any comment. I think we can tell from the name of the routine+platform now.

File: src/qibocal/protocols/characterization/couplers/coupler_qubit_spectroscopy.py [hunk 2]
```diff
@@ -76,34 +84,43 @@ def _acquisition(
...
+                Parameter.bias,
+                delta_bias_flux_range,
+                qubits=couplers,
+                type=SweeperType.OFFSET,
+            )
+        ]
 
     data = CouplerSpectroscopyData(
         resonator_type=platform.resonator_type,
+        flux_pulses=params.has_flux_params,
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/couplers/coupler_qubit_spectroscopy.py:109]:
```suggestion
        flux_pulses=params.flux_pulses,
```

File: src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py [hunk 3]
```diff
@@ -78,34 +76,44 @@ def _acquisition(
...
-    )
-
-    # This sweeper is implemented in the flux pulse amplitude and we need it to be that way.
-    sweeper_bias = Sweeper(
-        Parameter.bias,
-        delta_bias_range,
-        couplers=couplers,
-        type=SweeperType.ABSOLUTE,
-    )
+    if params.has_flux_params:
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py:79]:
```suggestion
    if params.flux_pulses:
```

File: src/qibocal/protocols/characterization/couplers/coupler_qubit_spectroscopy.py [hunk 4]
```diff
@@ -76,34 +84,43 @@ def _acquisition(
...
-    )
-
-    # This sweeper is implemented in the flux pulse amplitude and we need it to be that way.
-    sweeper_bias = Sweeper(
-        Parameter.bias,
-        delta_bias_range,
-        couplers=couplers,
-        type=SweeperType.ABSOLUTE,
-    )
+    if params.has_flux_params:
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/couplers/coupler_qubit_spectroscopy.py:87]:
```suggestion
    if params.flux_pulses:
```

File: src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py [hunk 5]
```diff
@@ -78,34 +76,44 @@ def _acquisition(
...
+                Parameter.bias,
+                delta_bias_flux_range,
+                qubits=couplers,
+                type=SweeperType.OFFSET,
+            )
+        ]
 
     data = CouplerSpectroscopyData(
         resonator_type=platform.resonator_type,
+        flux_pulses=params.has_flux_params,
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py:102]:
```suggestion
        flux_pulses=params.flux_pulses,
```

File: src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py [hunk 6]
```diff
@@ -167,15 +175,35 @@ def _plot(
...
+            #                 f"Coupler activation [{bias_flux_unit}]",
+            #                 f"Coupler sweetspot [{bias_flux_unit}]",
+            #             ],
+            #             [
+            #                 np.round(fit.pulse_amp[target], 4),
+            #                 np.round(fit.sweetspot[target], 4),
+            #             ],
+            #         )
+            #     )
+            # return [fig], fitting_report
```
Comment (GabrielePalazzo) [inline src/qibocal/protocols/characterization/couplers/coupler_resonator_spectroscopy.py:205]:
```suggestion
```

**Review by GabrielePalazzo — CHANGES_REQUESTED** [review]

**Review by GabrielePalazzo — APPROVED** [review]

**Comment by andrea-pasquale** [comment]
Could you please merge `main` here @Jacfomg?

File: src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py [hunk 7]
```diff
@@ -182,16 +183,33 @@ def create_flux_pulse_sweepers(
...
-        )
+        if isinstance(qubit, Coupler):
+            # FIXME: Missmatch with create_coupler_pulse and create_qubit_flux_pulse
+            pulse = platform.create_coupler_pulse(
+                qubit.name,
+                start=0,
+                duration=sequence.duration,
+                amplitude=1,
+            )
+            qubit = qubit.name
```
[the remark this answers is not in the document]
Comment (Jacfomg) [reply src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py:194]: It was already there, but due to they asking for bias sweepers it needed to follow the same approach as in #691 to reproduce the initial desired behavior.

**Review by andrea-pasquale — APPROVED** [review]

**Pull request merged**